src/storage/v2/disk/label_index.cpp

```diff
@@ -0,0 +1,209 @@
+  disk_transaction->SetCommitTimestamp(commit_ts);
+  auto status = disk_transaction->Commit();
+  if (!status.ok()) {
+    spdlog::error("rocksdb: {}", status.getState());
+  }
+  return status.ok();
+}
+
+}  // namespace
+
+DiskLabelIndex::DiskLabelIndex(Indices *indices, Constraints *constraints, const Config &config)
+    : LabelIndex(indices, constraints, config) {
+  kvstore_ = std::make_unique<RocksDBStorage>();
+  utils::EnsureDirOrDie(config.disk.label_index_directory);
+  kvstore_->options_.create_if_missing = true;
+  kvstore_->options_.comparator = new ComparatorWithU64TsImpl();
+  logging::AssertRocksDBStatus(rocksdb::TransactionDB::Open(kvstore_->options_, rocksdb::TransactionDBOptions(),
+                                                            config.disk.label_index_directory, &kvstore_->db_));
+}
+
+bool DiskLabelIndex::CreateIndex(LabelId label, const std::vector<std::pair<std::string, std::string>> &vertices) {
+  if (!index_.emplace(label).second) {
+    return false;
+  }
+
+  auto disk_transaction = CreateRocksDBTransaction();
+  for (const auto &[key, value] : vertices) {
+    disk_transaction->Put(key, value);
+  }
+  return CommitWithTimestamp(disk_transaction.get(), 0);
+}
+
+std::unique_ptr<rocksdb::Transaction> DiskLabelIndex::CreateRocksDBTransaction() const {
+  return std::unique_ptr<rocksdb::Transaction>(
+      kvstore_->db_->BeginTransaction(rocksdb::WriteOptions(), rocksdb::TransactionOptions()));
+}
+
+std::unique_ptr<rocksdb::Transaction> DiskLabelIndex::CreateAllReadingRocksDBTransaction() const {
+  auto tx = CreateRocksDBTransaction();
+  tx->SetReadTimestampForValidation(std::numeric_limits<uint64_t>::max());
+  return tx;
+}
+
+bool DiskLabelIndex::SyncVertexToLabelIndexStorage(const Vertex &vertex, uint64_t commit_timestamp) const {
+  auto disk_transaction = CreateRocksDBTransaction();
+
+  if (auto maybe_old_disk_key = utils::GetOldDiskKeyOrNull(vertex.delta); maybe_old_disk_key.has_value()) {
+    if (!disk_transaction->Delete(maybe_old_disk_key.value()).ok()) {
+      return false;
+    }
+  }
+
+  for (const LabelId index_label : index_) {
+    if (utils::Contains(vertex.labels, index_label)) {
+      if (!disk_transaction
+               ->Put(utils::SerializeVertexAsKeyForLabelIndex(index_label, vertex.gid),
+                     utils::SerializeVertexAsValueForLabelIndex(index_label, vertex.labels, vertex.properties))
+               .ok()) {
+        return false;
+      }
+    }
+  }
+
+  return CommitWithTimestamp(disk_transaction.get(), commit_timestamp);
+}
+
+/// TODO: this can probably be optimized
+bool DiskLabelIndex::ClearDeletedVertex(std::string_view gid, uint64_t transaction_commit_timestamp) const {
+  auto disk_transaction = CreateAllReadingRocksDBTransaction();
+
+  rocksdb::ReadOptions ro;
+  std::string strTs = utils::StringTimestamp(std::numeric_limits<uint64_t>::max());
+  rocksdb::Slice ts(strTs);
+  ro.timestamp = &ts;
```

Comment:
You are using this on few places, maybe put this in function? Also, can you explain logic, why do you set `strTs` to `std::numeric_limits<uint64_t>::max()`?